Assets/Scripts/R&D Tree/ResearchNode.cs

```diff
@@ -0,0 +1,106 @@
+using System.Runtime.CompilerServices;
+using UnityEngine;
+using UnityEngine.Events;
+using UnityEngine.UI;
+
+public class ResearchNode : MonoBehaviour
+{
+    [SerializeField] private RDTreeManager _treeManager;
+    public float ResearchValue { get; private set; }
+
+    public bool IsResearchInQueue { get; private set; } 
+
+    public ResearchNodeSetting NodeSetting {  get; private set; }
+
+    public ResearchStates CurrentResearchState { get;  private set; }
+
+    [HideInInspector]public UnityEvent ResearchDoneEvent;
+
+    private void Start()
+    {
+        NodeSetting = GetComponent<ResearchNodeSetting>();
+        _treeManager.AllNodes.Add(this);    
```

Comment:
Directly accessing another class's variable. Make a separate function in the class to make this clean.